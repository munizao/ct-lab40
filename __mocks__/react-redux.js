jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ user: {}, repos: {} })),
  };
});
