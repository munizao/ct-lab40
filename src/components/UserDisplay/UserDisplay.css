.UserDisplay {
  width: 25vw;
  img {
    width: 5vw;
  }
}