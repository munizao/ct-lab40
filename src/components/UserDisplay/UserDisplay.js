import React from 'react';
import { useSelector } from 'react-redux';
import { selectUser, selectLoading } from '../../redux/selectors/user-selector';
import styles from './UserDisplay.css';

const UserDisplay = () => {
  const loading = useSelector(selectLoading);
  const loadingDOM = loading && <span>Loading...</span>;
  const userInfo = useSelector(selectUser);
  const userInfoDOM = userInfo &&
  <>
    <div>{userInfo.avatar_url && <img src={userInfo.avatar_url} />}</div>
    <p><a href={userInfo.html_url}>{userInfo.username}</a></p>
    <p>Followers: {userInfo.followers}</p>
    <p>Following: {userInfo.following}</p>
    <p>Public Repos: <a href={userInfo.repos_url}>{userInfo.public_repos}</a></p>
  </>;

  return (
    <div className={styles.UserDisplay}>
      {loadingDOM}
      {userInfoDOM}
    </div>
  );
};

export default UserDisplay;
