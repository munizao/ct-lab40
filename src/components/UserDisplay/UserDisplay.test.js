import React from 'react';
import 'react-redux';
import { shallow } from 'enzyme';
import UserDisplay from './UserDisplay';

describe('UserDisplay component', () => {
  it('renders UserDisplay', () => {
    const wrapper = shallow(<UserDisplay />);
    expect(wrapper).toMatchSnapshot();
  });
});
