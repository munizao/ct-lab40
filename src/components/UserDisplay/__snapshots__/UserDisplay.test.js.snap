// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay component renders UserDisplay 1`] = `
<div
  className="UserBar"
/>
`;
